client/my-sites/plans-features-main/hooks/use-plan-types-with-intent.ts

```diff
@@ -0,0 +1,130 @@
+import {
+	TYPE_BLOGGER,
+	TYPE_BUSINESS,
+	TYPE_ECOMMERCE,
+	TYPE_ENTERPRISE_GRID_WPCOM,
+	TYPE_FREE,
+	TYPE_PERSONAL,
+	TYPE_PREMIUM,
+	getPlan,
+	isBloggerPlan,
+} from '@automattic/calypso-products';
+import { useSiteIntent } from '@automattic/data-stores';
+import { useSelector } from 'react-redux';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export type Intent =
+	| 'blog-onboarding'
+	| 'newsletter'
+	| 'link-in-bio'
+	| 'new-hosted-site'
+	| 'new-hosted-site-hosting-flow'
+	| 'plugins'
+	| 'jetpack-app' // newly added
+	| 'plans-import'
+	| 'plans-ecommerce'
+	| 'default';
+
+interface Props {
+	intent?: Intent | null | undefined;
+	selectedPlan: string;
+	sitePlanSlug: string;
+	hideEnterprisePlan: boolean;
+	currentSitePlanSlug?: string;
+}
+
+interface PlanTypesWithIntent {
+	intent: Intent;
+	planTypes: string[];
+}
+
+const useIntentFromSiteMeta = (): Intent | null => {
+	const selectedSiteId = useSelector( getSelectedSiteId ) ?? undefined;
+	const intent = useSiteIntent( selectedSiteId );
```

Comment:
Need to follow up on conclusions from p1685550995632009-slack-C03N25JPCE4  as there seems to have been some issues with endpoint used in a similar hook in ETK. [Here](https://github.com/Automattic/wp-calypso/pull/77629#discussion_r1212467275) there's chat about deprecating the data-stores' one (this) as well.